command/job_init.go

```diff
@@ -90,27 +101,82 @@ func (c *JobInitCommand) Run(args []string) int {
 		c.Ui.Error(fmt.Sprintf("Failed to stat '%s': %v", filename, err))
 		return 1
 	}
-	if !os.IsNotExist(err) {
+	if !os.IsNotExist(err) && !listTemplates {
 		c.Ui.Error(fmt.Sprintf("Job '%s' already exists", filename))
 		return 1
 	}
 
 	var jobSpec []byte
-	switch {
-	case connect && !short:
-		jobSpec, err = Asset("command/assets/connect.nomad")
-	case connect && short:
-		jobSpec, err = Asset("command/assets/connect-short.nomad")
-	case !connect && short:
-		jobSpec, err = Asset("command/assets/example-short.nomad")
-	default:
-		jobSpec, err = Asset("command/assets/example.nomad")
-	}
-	if err != nil {
-		// should never see this because we've precompiled the assets
-		// as part of `make generate-examples`
-		c.Ui.Error(fmt.Sprintf("Accessed non-existent asset: %s", err))
-		return 1
+
+	if listTemplates {
+		// Get the HTTP client
+		client, err := c.Meta.Client()
+		qo := &api.QueryOptions{
+			Namespace: c.Meta.namespace,
+		}
+		// Get and list all variables at nomad/job-templates
+		vars, _, err := client.Variables().PrefixList("nomad/job-templates/", qo)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error retrieving vars: %s", err))
+			return 1
+		}
+
+		if len(vars) == 0 {
+			c.Ui.Error(fmt.Sprintf("No variables in nomad/job-templates"))
+			return 1
+		} else {
+			c.Ui.Output(fmt.Sprintf("Use nomad job init -template=<template> with any of the following:"))
+			for _, v := range vars {
+				fmt.Println(strings.TrimPrefix(v.Path, "nomad/job-templates/"))
```

Comment:
We should use `c.Ui.Output` here so that we know it's all going thru that interface. Also, do we want to indent this slightly (like 2 spaces?) to make it stand out?